internal/controller/controllers/events_uploader_handler.go

```diff
@@ -0,0 +1,319 @@
+package controllers
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+	"strings"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type serializedAuthMap struct {
+	Auths map[string]serializedAuth `json:"auths"`
+}
+type serializedAuth struct {
+	Auth string `json:"auth"`
+}
+
+type uploader struct {
+	db               *gorm.DB
+	log              log.FieldLogger
+	client           k8sclient.K8SClient
+	hasOCMPullSecret bool
+	enabled          bool
+	endpoint         string
+}
+
+const (
+	openshiftTokenKey = "cloud.openshift.com"
+)
+
+func NewEventsUploaderWrapper(db *gorm.DB, log logrus.FieldLogger, client k8sclient.K8SClient, enabled, hasOCMPullSecret bool, endpoint string) *uploader {
+	return &uploader{
+		db:               db,
+		log:              log,
+		client:           client,
+		hasOCMPullSecret: hasOCMPullSecret,
+		enabled:          enabled,
+		endpoint:         endpoint,
+	}
+}
+
+func (u *uploader) IsEnabled(ctx context.Context) bool {
+	if u.hasOCMPullSecret && u.enabled {
```

Comment:
I wonder if `hasOCMPullSecret` is really useful.
We could just check all the time (assuming `client` is set) and only disable if we find it and the cloud auth is missing, right?

Reply:
Modified the logic to reflect this! May I have your input on how it's implemented?

[`getAuthToken`](https://github.com/openshift/assisted-service/pull/4880/files#diff-be3c5c34d4670110a2ffa6cd391d344a1c14941e7813cce8eb05cd9e92faf686R43) is run every time and [`getOCMPullSecret`](https://github.com/openshift/assisted-service/pull/4880/files#diff-be3c5c34d4670110a2ffa6cd391d344a1c14941e7813cce8eb05cd9e92faf686R29) checks if the OCM pull secret exists (and returns it if it does)

Whenever the OCM pull secret doesn't exist, I'm assuming that it's not an OCP cluster and we can safely use the cluster's pull secret.